### DISCUSSION TITLE
Does the 'Showers' variable always represent rain?

### DISCUSSION THREAD

**Discussion by colinnuk** [open-meteo/open-meteo#717] [Q&A] [answered]
Question about the showers variable - does it always represent rain rather than snow? Or could showers be _either_ rain or snow, depending on temperature? I want to add showers into either 'rain' or 'snow' so I get the total precip in either form.

I've had a look thru the source code quickly - as far as I can tell you are normally able to source 'showers' directly from the model, so I'm not sure exactly what this represents. Perhaps it differs depending on the model?

**Comment by patrick-zippenfenig** [comment]
Technically showers could be snow, but convective snow is quite rare. Most models do not even separate between showers and rain in the first place. Also, snowfall data is not very reliable. I just spend 3 days to correct snowfall amount for the ICON weather models which forecast snowfall at 8°C while they even show a snowfall height of 800 metre above ground.

**Reply by colinnuk** [reply]
OK thanks for replying! Convective snow isn't rare in the mountains in western Canada - particularly in spring.

I think elsewhere you'd mentioned you were thinking of changing some parameters to split rain & snow by temperature rather than taking it from each model - I guess that sort of system would put the precip into the most 'likely' bucket.

**Reply by patrick-zippenfenig** [reply, marked as answer]
Yes, I noticed that some weather models predict snow at >8°C. This is quite unlikely. I added a post-processing step that converts snow back to rain if temperature is above 1.5°C.

Canada is covered by the HRRR model as well as HRDPS. At least HRRR is a convection allowing model and does not separate between large scale and convective precipitation. HRRR does not contain a showers field in its model output -> [showers will always be 0](https://github.com/open-meteo/open-meteo/blob/main/Sources/App/Gfs/GfsController.swift#L166).

**Comment by bonnells** [comment]
In Australia snow is largely confined to the Alpine region and hence has an essentially commercial focus. As such there isn't the same importance as there is in other parts of the world. I obtained some 'Open-Meteo' data during the recent season, which was a poor one, simply because I was interested in some ideas related to my past dealings with trying to forecast it. I am sure there are more accomplished operators elsewhere. Nevertheless I learned a few things which might be of note to those of you who have an interest in this aspect of the weather.


Whether it rains or snows is of course a function of the atmospheric warmth. However it isn't simply a factor of the temperature, other issues come to play. Focusing solely on the surface conditions here, the humidity is also an important ingredient when the temperature hovers just above zero. The lower the humidity, the more cooling from evaporation/sublimation that arises as the snow flakes are entrained into a drier environment. The article: [https://www.diva-portal.org/smash/get/diva2:968860/FULLTEXT01.pdf](url) deals with these aspects. Much of this will probably be of little or no interest here, but Figure 3 from the article is reproduced below. It gives a relationship between the wet-bulb temperature and the conditional probability of any precipitation being snow. Such a relationship as this, a sigmoid curve, represents a logistic distribution. One can quite accurately estimate this curve from the plot and I came up with the following relationship:

`	logit(prob) = 3.93265 - 3.444 * T(wet)
`

![twet_logistic](https://github.com/user-attachments/assets/6c71422c-1a3f-4d2c-83a0-17976a1ed25f)

One can convert this into an actual probability by simply inserting the value of the wet-bulb temperature into the equation and exponentiating the result. If one takes this value and divides it by the the very same value with 1 added to it, they get the conditional probability that any precipitation will fall as snow, rather than rain. For example, suppose the wet-bulb temperature is +1. The logit becomes 3.93265-3.444*1 = 0.489; exp(0.489) = 1.63 and 1.63/(1+1.63) = 0.62; i.e. all things being equal, there is a 62% chance it will be snow as opposed to rain.

So anybody using the temperature/humidity data here to infer whether or not it is raining or snowing should focus on the wet-bulb temperature. I raised this recently and Patrick responded with some information that will be of interest in relation to all of this.
[https://github.com/open-meteo/open-meteo/discussions/1091](url)

I will have more to say on this matter in a few days time.

Dave

**Reply by patrick-zippenfenig** [reply]
Many weather models use a threshold of ~1.3°C wet bulb temperature to parametrise snowfall. It is also important to analyse the vertical column to differentiate different precipitation types.

Luckily most weather models offer precipitation type or snowfall amount directly. For others I am considering to build a post processing step to improve it (https://github.com/open-meteo/open-meteo/issues/205

ECMWF IFS just made precipitation type available which can now be used to better differentiate snowfall: https://github.com/open-meteo/open-meteo/pull/1117

**Comment by bonnells** [comment]
A weather resource that could be used in conjunction with the data available here can be found at the University of Wyoming:
[http://weather.uwyo.edu/](url)

Of specific interest in this post is the surface observations: [http://weather.uwyo.edu/surface/meteorogram/index.shtml](url)

For example, the current 'raw' observations (past 24hrs) for Berlin [EDDB] are:
[http://weather.uwyo.edu/cgi-bin/wyowx.fcgi?TYPE=metar&DATE=current&HOUR=current&UNITS=M&STATION=EDDB](url)

There is a 'listing' option, where the same raw message is decoded:
[http://weather.uwyo.edu/cgi-bin/wyowx.fcgi?TYPE=sflist&DATE=current&HOUR=current&UNITS=M&STATION=EDDB](url)

To see what the airport code is for various locations across the region of the world on each specific page, view the page source. 
[Right click the page and click 'view page source']

The weather observations here are from those taken for aviation purposes. There is a specific structure to the way the observations are coded and I can say a few things later for those who are interested and unaware. In recent weeks I have grabbed some historical data from the USA, where the temperature data is available to the nearest 0.1 deg and hence is better resolved than elsewhere where the temperature and dew point are rounded to the degree, as well as Europe. From this I could isolate occasions when it rained and when it snowed and derive the wet-bulb temperature [see: [https://github.com/open-meteo/open-meteo/discussions/1091](url) ] at that time. This then allows one to derive a logistic regression fit to the wet-bulb temperature which allows one to assign a probability to the possibility of it snowing, as opposed to raining, for that temperature/humidity combination.

Patrick has informed us that a wet-bulb temperature of +1.3 degrees is used to distinguish between the two. [1] Personally I think this
is a bit too high and [2] a more informative approach would be to assign a probability to it, rather than a straight out 'snow or rain'.

Combining the USA data for 6 locations I obtained a logit:

`logit(prb) = 2.30472-2.62276 x t(wet)`

whilst for the European data:

`logit(prb) = 0.95157-2.20695 x t(wet)`

The actual probability can be derived by transforming the logit; `prb = exp(logit(p))/(1+exp(logit(p))`. It should be clear that if the logit equals 0, the probability = 0.5 because exp(0) = 1. From the above logits, the USA data suggests that a t(wet) = 2.30472/2.62276 = 0.87 equates to a 50/50% chance of snow, whilst for the European data, t(wet) = 0.95157/2.20695 = 0.43. Both of these are a good deal lower than the previously mentioned +1.3. In fact, inserting 1.3 into the 2 logits gives a probability of around 25% for the USA data and 13% for Europe.

A plot of how these probabilities vary with the wet-bulb temperature, along with the fit to the data I presented in an earlier contribution here is given below. The * refers to the 50/50 chance @ +1.3 as suggested previously. It is clearly higher than indicated in these 3 curves.

![snow_vs_rain_logistic](https://github.com/user-attachments/assets/3203752a-887b-49af-a2d5-b936c2815d0d)

One might note that the larger the slope term in the logit equation, the sharper the delineation between snow and rain and hence the reason why all 3 curves approach the same probabilities when t(wet) reaches +2 despite being quite divergent between 0 and +1. ['Olsen' refers to the plotted data in my previous contribution.]

Those of you who wish to give some idea of the chances of snow or rain should consider using a logistic fit to the wet-bulb temperature derived from the temperature & humidity, as Patrick has indicated is available. These probabilities are known as 'conditional probabilities', being conditional on it precipitating in the first place. So if the probability of precipitation was 50%, the chances of snow are half that which is derived here. If it was 80% it would be 4/5th's etc. Given the variation suggested by these curves it might lead to further analysis/discussion as to what might be a good relationship.  Maybe the USA/Europe data should be pooled together to give one fit? Maybe I should grab a third data set and verify it against these relationships? There will always be some uncertainty as it will ultimately depend on such aspects as the atmospheric conditions immediately above the surface and how heavy the precipitation falls, as this will determine how quickly the precipitation melts etc.

I accumulated some forecasts for the Alpine region in Australia across recent months. I am starting to collate it, amongst other things, and plan to pass on some thoughts on it to Patrick in the future.

Dave
